### PR TITLE
app-admin/syslog-ng: fix automagic of dev-libs/grpc

### PR DESCRIPTION
**Pull request by sevz17**

### CODE REVIEW

**Comment by hydrapolic** [comment]
```bash
--- /var/db/repos/gentoo/app-admin/syslog-ng/syslog-ng-4.4.0-r1.ebuild  2023-10-18 17:10:34.000000000 +0000
+++ syslog-ng-4.4.0-r2.ebuild   2024-01-03 14:04:35.102296685 +0000
@@ -13,7 +13,7 @@
 LICENSE="GPL-2+ LGPL-2.1+"
 SLOT="0"
 KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~loong ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86"
-IUSE="amqp caps dbi geoip2 http json kafka mongodb pacct python redis smtp snmp test spoof-source systemd tcpd"
+IUSE="amqp caps dbi geoip2 grpc http json kafka mongodb pacct python redis smtp snmp test spoof-source systemd tcpd"
 REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )
        test? ( python )"
 RESTRICT="!test? ( test )"
@@ -28,6 +28,10 @@
        caps? ( sys-libs/libcap )
        dbi? ( >=dev-db/libdbi-0.9.0 )
        geoip2? ( dev-libs/libmaxminddb:= )
+       grpc? (
+               dev-libs/protobuf:=
+               net-libs/grpc:=
+       )
        http? ( net-misc/curl )
        json? ( >=dev-libs/json-c-0.9:= )
        kafka? ( >=dev-libs/librdkafka-1.0.0:= )
@@ -46,10 +50,13 @@
        tcpd? ( >=sys-apps/tcp-wrappers-7.6 )"
 DEPEND="${RDEPEND}
        test? ( dev-libs/criterion )"
+# uses protoc (dev-libs/protobuf)
 BDEPEND="
+       grpc? ( dev-libs/protobuf:= )
        >=sys-devel/bison-3.7.6
        sys-devel/flex
-       virtual/pkgconfig"
+       virtual/pkgconfig
+"
 
 DOCS=( AUTHORS NEWS.md CONTRIBUTING.md contrib/syslog-ng.conf.{HP-UX,RedHat,SunOS,doc}
        contrib/syslog2ng "${T}/syslog-ng.conf.gentoo.hardened"
@@ -126,10 +133,12 @@
                --with-python-packages=none
                --with-systemdsystemunitdir="$(systemd_get_systemunitdir)"
                $(use_enable amqp)
-               $(usex amqp --with-librabbitmq-client=system --without-librabbitmq-client)
+               $(use_with amqp librabbitmq-client system)
                $(use_enable caps linux-caps)
                $(use_enable dbi sql)
                $(use_enable geoip2)
+               $(use_enable grpc)
+               $(use_enable grpc cpp)
                $(use_enable http)
                $(use_enable json)
                $(use_enable kafka)
```

**Review by hydrapolic — APPROVED** [review]

File: app-admin/syslog-ng/syslog-ng-4.4.0-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+	)
+	redis? ( >=dev-libs/hiredis-0.11.0:= )
+	smtp? ( net-libs/libesmtp:= )
+	snmp? ( net-analyzer/net-snmp:0= )
+	spoof-source? ( net-libs/libnet:1.1 )
+	systemd? ( sys-apps/systemd:= )
+	tcpd? ( >=sys-apps/tcp-wrappers-7.6 )"
+DEPEND="${RDEPEND}
+	test? ( dev-libs/criterion )"
+# uses protoc (dev-libs/protobuf)
```
[the remark this answers is not in the document]
Comment (sevz17) [reply app-admin/syslog-ng/syslog-ng-4.4.0-r2.ebuild]: It meant that not only links against libprotobuf but also uses the protoc tool, I just dropped the comment

**Comment by Flowdalic** [comment]
```
 * VDB: detected possibly incorrect RDEPEND (app-admin/syslog-ng-4.4.0-r2)
 * dev-libs/libpcre2 | dev-libs/libpcre2:=
 *                   > net-libs/paho-mqtt-c:1.3
 ```

There seems to be another automagic dependency wrt to MQTT: https://github.com/search?q=repo%3Asyslog-ng%2Fsyslog-ng%20paho-mqtt&type=code

@sevz17 @hydrapolic could I bribe you into fixing this as well?

**Comment by sevz17** [comment]
> ```
>  * VDB: detected possibly incorrect RDEPEND (app-admin/syslog-ng-4.4.0-r2)
>  * dev-libs/libpcre2 | dev-libs/libpcre2:=
>  *                   > net-libs/paho-mqtt-c:1.3
> ```
> 
> There seems to be another automagic dependency wrt to MQTT: https://github.com/search?q=repo%3Asyslog-ng%2Fsyslog-ng%20paho-mqtt&type=code
> 
> @sevz17 @hydrapolic could I bribe you into fixing this as well?

Sure, I'll take a look the Saturday.

**Pull request closed without merging**